src/plugin/class-liberate-controller.php

```diff
@@ -22,6 +22,31 @@ public function get_storage_post_type(): string {
 		return $this->storage_post_type;
 	}
 
+	public function valid_request_for_insert( $request ): bool|WP_Error {
+		global $wpdb;
+
+		$request_data = json_decode( $request->get_body(), true );
+		$guid         = $request_data['sourceUrl']; // required arg, will always be present at this point
```

Comment:
The comment makes me think we're validating the `sourceUrl` elsewhere but I couldn't find where we're doing that validation. Could you point me to where that validation is happening?

Reply:
That comment intends to clarify we don't need to check for existence of the `sourceUrl` key in `$request_data` since [it's defined as required in schema](https://github.com/WordPress/try-wordpress/blob/8681178f7e08195b2324555d3ca3ff4a6b607421/src/plugin/class-blogpost-controller.php#L108) and validation accounts for that. So, if we got past that, this field would always exist in the array.

Reply:
Thanks, I assumed I was missing something :)